### ISSUE TITLE
Snap Package Support (Feature Request)

### ISSUE THREAD

**Issue by mikeroyal** [openfaas/faas-cli#569]
I was thinking in the install and setup could be improved for Ubuntu and other Linux distros by having a [snap package](https://snapcraft.io/) version. If it's not a priority would you mind if I made the snap?

[Crafting your first snap](https://tutorials.ubuntu.com/tutorial/create-your-first-snap#0)

[Build and test snap](https://snapcraft.io/build)

[Getting Started snap template](https://github.com/snapcrafters/fork-and-rename-me)

**Comment by burtonr** [comment]
Hey @mikeroyal Although I personally use Ubuntu (and snaps) primarily, this isn't a priority for the project. 

The challenge is in maintaining the versions with releases.

We currently only officially support `brew` installer on MacOS as that is the primary platform for our users. Even that can be a challenge to keep up with the changes.

If you could propose a way that the maintenance of the snap would be automated, and reliable, it may be something that could be considered. However, as of right now, I don't think this will come to fruition. 

I don't mean to knock your idea. I do like the idea, it's just difficult to keep up with multiple installers in the long term...

**Comment by mikeroyal** [comment]
@alexellis, Yes, here are the docs if you want to read more in-depth. https://microk8s.io/docs/

**List of available addons**

 **dns**: Deploy kube dns. This addon may be required by others thus we recommend you always enable it.

**dashboard**: Deploy kubernetes dashboard as well as grafana and influxdb. To access grafana point your browser to the url reported by **microk8s.kubectl cluster-info**.

**storage**: Create a default storage class. This storage class makes use of the hostpath-provisioner pointing to a directory on the host. Persistent volumes are created under **${SNAP_COMMON}/default-storage**. Upon disabling this addon you will be asked if you want to delete the persistent volumes created.

 **ingress**: Create an ingress controller.

 **gpu**: Expose GPU(s) to MicroK8s by enabling the nvidia-docker runtime and nvidia-device-plugin-daemonset. Requires NVIDIA drivers to already be installed on the host system.

 **istio**: Deploy the core [Istio](https://istio.io/) services. You can use the **microk8s.istioctl** command to manage your deployments.

**registry**: Deploy a docker private registry and expose it on **localhost:32000**. The storage addon will be enabled as part of this addon. To[use the registry](https://microk8s.io/docs/registry) you can use the microk8s.docker command.

**metrics-server**: Deploy the [Metrics Server](https://kubernetes.io/docs/tasks/debug-application-cluster/core-metrics-pipeline/#metrics-server).

**Comment by alexellis** [comment]
Thanks so much for the info. Do you want to join Slack to talk more? https://docs.openfaas.com/community/

How do we go about writing a test snap for this?

Alex

**Comment by alexellis** [comment]
@mikeroyal

**Comment by Roadmaster** [comment]
@mikeroyal hello there, 

You currently own the faas-cli snap name, but we (the Snap Store team) received a request from the openfaas project to get ownership of the name. I wonder if you could confirm whether you agree with this in general (@alexellis is the requester).

I will also need to email you privately to confirm (as I can't verify your Snap Store email address matches your github identity) but it'll speed things up if you and @alexellis decide here whether it's OK to do the transfer.

- Daniel

**Comment by mikeroyal** [comment]
Yeah, that's fine @Roadmaster

**Comment by alexellis** [comment]
Thank you both 👍

**Comment by alexellis** [comment]
Hi @mikeroyal I can see some releases and snapshots, we'd assumed there was no package and started work on this to resolve, but it looks like you had something up.

In the dashboard I see this error though

<img width="779" alt="Screenshot 2020-01-15 at 22 30 41" src="https://user-images.githubusercontent.com/6358735/72477078-afd1f600-37e6-11ea-8e86-e193502ade7a.png">

**Comment by Roadmaster** [comment]
The above is because Mike's snap uses the docker interface which needs to be explicitly approved (you can request it in the forum: https://forum.snapcraft.io/c/store-requests.

Alternatively, if you make it so that the snap doesn't require use of docker, it would likely pass the automatic review process.

- Daniel
